idaes/power_generation/carbon_capture/mea_solvent_system/property_package/liquid_prop.py

```diff
@@ -0,0 +1,1282 @@
+from pyomo.common.config import  ConfigValue, In
+from pyomo.opt import SolverFactory
+
+# Import IDAES libraries
+from idaes.core import (declare_process_block_class,
+                        MaterialFlowBasis,
+                        PhysicalParameterBlock,
+                        StateBlockData,
+                        StateBlock,
+                        Component,
+                        LiquidPhase)
+
+from idaes.core.util.initialization import (fix_state_vars,
+                                            revert_state_vars,
+                                            solve_indexed_blocks)
+from idaes.core.util.misc import add_object_reference
+from idaes.core.util.model_statistics  import (degrees_of_freedom,
+                                              number_unfixed_variables,
+                                              report_statistics)
+import idaes.logger as idaeslog
+
+__author__ = "Paul Akula, John Eslick"
+
+# Set up logger
+_log = idaeslog.getLogger(__name__)
+
+@declare_process_block_class("LiquidParameterBlock")
+class PhysicalParameterData(PhysicalParameterBlock):
+    """
+    Liquid Phase Property Parameter Block Class
+
+    Contains parameters and indexing sets associated with
+    liquid phase properties for amine-based scrubbing process.
+
+    """
+    CONFIG = PhysicalParameterBlock.CONFIG()
+    CONFIG.declare("process_type", ConfigValue(
+        default='Absorber',
+        domain=In(['Absorber', 'Stripper','Regenerator']),
+        description="Flag indicating the type of  process",
+        doc="""Flag indicating either absoprtion or stripping process.
+               Heat of absorption depends on teh process type """))
+
+    def build(self):
+        '''
+        Callable method for Block construction.
+        '''
+        super(PhysicalParameterData, self).build()
+
+        self._state_block_class = LiquidStateBlock
+
+        # Create Phase object
+        self.Liq = LiquidPhase()
+
+        # Create Component objects
+        self.MEA = Component()
+        self.CO2 = Component()
+        self.H2O = Component()
+
+        #create component list for true species
+        self.component_list_true = Set(initialize=
+                             ['CO2', 'H2O', 'MEA','MEA+','MEACOO-', 'HCO3-'])
+        self.component_list_anion = Set(initialize=['MEACOO-', 'HCO3-'])
+        self.component_list_cation = Set(initialize=['MEA+'])
+        self.component_list_solvent = Set(initialize=['H2O', 'MEA'])
+        self.component_list_diffus = Set(initialize=
+                                          ['CO2','MEA','MEA+','MEACOO-'])
+        #Reaction index
+        self.reaction_idx = Set(initialize=['R1','R2'])
+
+        # Thermodynamic reference state
+        self.pressure_ref = Param(within=PositiveReals,
+                                  mutable=True,
+                                  default=101325,
+                                  doc='Reference pressure [Pa]')
```

Comment:
The imported Pyomo unit is only used in the default unit setting of the metadata object which is the SI unit. All property units listed in the property methods are in SI units so there will be no inconsistency.

Reply:
Whilst there may not be an actual inconsistency in the calculations, it will not appear that way to Pyomo unless all `Vars` and `Param` (in all property packages and unit models) have units attached to them.

The metadata is used primarily by the control volumes to determine the units of any `Vars` they need to create, and these are attached ot the `Vars` as they are created. However, as far as Pyomo is concerned, all the `Vars` you created here are unitless - thus, if you were to do a unit consistency check it would fail.

Unfortunately, units in Pyomo is and all-or-nothing approach. Unless you specify units for everything, you might as well specify none as you won't be able to use them if you don't.

Reply:
The 'units'  keyword is now used to set units for all  ```Var``` and ```Param``` created.  Changes will appear in the next push.